src/Raylib.php

```diff
@@ -931,6 +961,40 @@ public function loadFont(string $filename): Types\Font
         );
     }
 
+    /**
+     * @psalm-suppress UndefinedPropertyFetch
+     * @psalm-suppress InvalidPassByReference
+     * @psalm-suppress MixedArrayAssignment
+     * @psalm-suppress MixedArgument
+     * @psalm-suppress MixedAssignment
+     * @psalm-suppress MixedAssignment
+     * @psalm-suppress MixedPropertyFetch
+     */
+    public function loadFontEx(string $fileName, int $fontSize, int $fontChars, int $charsCount): Types\Font
+    {
+        $chars = FFI::addr($this->ffi->new('int32_t'));
```

Comment:
This line here is problematic. `LoadFontEx` expects a pointer to an integer so it can write into it, this line points to a single integer causing the program to read unexpected memory spaces.

[Checking raylib's code](https://github.com/raysan5/raylib/blob/3.5.0/src/text.c#L569-L574) I can tell that if this pointer is null, Raylib will automatically allocate and fill memory properly. So let's just ignore this parameter for now. Please replace this initial part of the code with the following:

```
$fontCharsPointer = $this->ffi->new('int*');

$font = $this->ffi->LoadFontEx($fileName, $fontSize, $fontCharsPointer, $charsCount);
```

Notice how `$fontChars` is completely ignored. If this becomes a problem someday, we'll just file a bug issue for it and investigate further.

Reply:
Thanks, for your help! It was now clear to me where the error was occurring. :smiley: